src/components/application_manager/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -1191,7 +1199,8 @@ bool RegisterAppInterfaceRequest::IsApplicationWithSameAppIdRegistered() {
       application_manager_.applications().GetData();
 
 #ifdef SDL_REMOTE_CONTROL
-  IsSameAppId matcher(mobile_app_id.AsMBString(),
+  const std::string tmp_mobile_app_id = mobile_app_id.AsMBString();
+  IsSameAppId matcher(tmp_mobile_app_id,
```

Comment:
@okozlovlux IsSameAppId holds the app_id as a string reference, but the time of life of the object that AsMBString() returns is not enough, so I create new object tmp_mobile_app_id that will live longer enough. Another approach to fix this problem is changing the IsSameAppId to hold the app_id not as a string reference, but as a string object.